src/main/java/com/bruincreates/server/service/ProductReviewService.java

```diff
@@ -1,13 +1,104 @@
 package com.bruincreates.server.service;
 
 import com.bruincreates.server.dao.mapper.ProductReviewMapper;
+import com.bruincreates.server.dao.po.ProductReview;
+import com.bruincreates.server.dao.po.ProductReviewExample;
+import com.bruincreates.server.model.request.CreateReviewRequest;
+import com.bruincreates.server.model.request.DeleteReviewRequest;
+import com.bruincreates.server.exception.BadRequestException;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Service;
 
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
 @Service
 public class ProductReviewService {
 
     @Autowired
     ProductReviewMapper productReviewMapper;
 
+    @Autowired
+    OrderService orderService;
+
+    public void createReview (CreateReviewRequest request) throws BadRequestException {
+
+        String username = request.getUsername();
+        String productId = request.getProductId();
+        String reviewContent = request.getProductReview();
```

Comment:
a too long review should give an exception (length is capped at 512 as defined in db.sql)